sirepo/pkcli/admin.py

```diff
@@ -77,44 +77,60 @@ def move_user_sims(target_uid=''):
         shutil.move(lib_file, target)
 
 
-def purge_guest_users(days=180, confirm=False):
-    """Remove old users from db which have not registered.
+def purge_free_users(days=180, confirm=False):
+    """Remove basic users simulations
 
     Args:
         days (int): maximum days of untouched files (old is mtime > days)
         confirm (bool): delete the directories if True (else don't delete) [False]
 
     Returns:
-        (list, list): dirs and uids of removed guest users (or to remove if confirm)
+        (list, list): dirs and uids of removed users (or to remove if confirm)
     """
 
+    def _check_expired_run_dir(path):
+        d = str(path).lower()
+        if 'report' in d or 'animation' in d:
+            for f in pkio.walk_tree(path):
+                # https://github.com/radiasoft/sirepo/issues/1888
+                if not f.exists():
+                    continue
+                if (n - n.fromtimestamp(f.mtime())).days <= days:
+                    return
+            else:
+                res.append(str(path))
+
+    def _get_premium_uids():
+        r = auth_db.UserRole
+        return [
+            x[0] for x in r.query.with_entities(r.uid).filter(
+                r.role == auth.ROLE_PREMIUM,
+            ).distinct().all()
+        ]
+
     days = int(days)
     assert days >= 1, \
         '{}: days must be a positive integer'
     server.init()
     from sirepo import srtime
 
-    guest_uids = auth.guest_uids()
-    now = srtime.utc_now()
-    dirs_and_uids = {}
-
-    for d in pkio.sorted_glob(simulation_db.user_dir_name().join('*')):
-        uid = simulation_db.uid_from_dir_name(d)
-        if _is_src_dir(d):
-            continue
-        if uid not in guest_uids:
-            continue
-        for f in pkio.walk_tree(d):
-            if (now - now.fromtimestamp(f.mtime())).days <= days:
-                break
-        else:
-
-            dirs_and_uids[d] = uid
-    if confirm:
-        pkio.unchecked_remove(*dirs_and_uids.keys())
-        auth_db.UserRegistration.delete_all_for_column_by_values('uid', dirs_and_uids.values())
-
-    return dirs_and_uids
+    with auth_db.thread_lock:
```

Comment:
This is insufficient and problematic. The problem is that Flask creates run directories without checking auth_db thread_lock, and in any event, flask is running in a different process than this pkcli so thread locks doesn't work.

We need to think about if locking is necessary when removing run directories. 

Since the supervisor controls run directories. We could send it a list of run directories to purge from this code.

The advantage of this is that the supervisor could actually store a different job state when a simulation is purged instead of missing. It would mean not hitting runDirNotFound on this case, that is, the supervisor would see the new state (e.g. job.PURGED_BASIC) and it would be able to return this to the GUI, which could then explain to the user: your simulations were purged, and you might want to upgrade to premium so this doesn't happen.